test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -1138,6 +1138,66 @@ static void test_out_ipv4_udp_esp_null_sha256(void)
 	ipsec_sa_destroy(sa);
 }
 
+static void test_debug_sa_update_seq_num(void)
```

Comment:
What is the purpose of this test?

According to the subject line of the commit ("add IPSEC SA sequence number test") this tests something about how SAs handle sequence numbers. Like maybe checking that sequence number starts from one and gets incremented by one for each packet. But it does not test that.

I did not understand the actual text of the commit message.

The name of the test function is ipsec_check_debug_sa_update_seq_num(), leading me to think that the test tests that odp_ipsec_debug_sa_update() somehow works. But it does not test that. odp_ipsec_debug_sa_update() could update the sequence number to some other value than requested or do nothing at all and the test would still pass.

So if the test is not about testing odp_ipsec_debug_sa_update() but about testing ODP IPsec functionality by using  dp_ipsec_debug_sa_update() as part of the test machinery, then the test function should be renamed.

But looking at the code, the only thing the test tests is that IPsec packet is succesfully received if the first received sequence number is 1235. That might test the anti-replay implementation just a little bit, except that the test does not set a non-zero value to antireplay_ws config parameter to ensure that anti-replay is enabled.

So maybe the only thing the test really tests is that the underlying ODP implementation has the SA debug API addition so that compilation does not fail? But I think a simpler test would suffice for that.

Reply:
> I did not understand the actual text of the commit message.

Ack. Updated the commit message to reflect the actual test being done in v3.


> odp_ipsec_debug_sa_update() could update the sequence number to some other value than requested or do nothing at all and the test would still pass.

Didn't realize this!. My original thought was to have a known vector (NULL cipher with SHA256 auth) which will have the sequence number as well embedded in the result packet. So the memcmp() operation will confirm that the sequence number is updated. But I ended up using AES-GCM here resulting in an incomplete test.

In v3, I have updated the test to check the sequence number present in the IPsec header of the resulting packet. Also I have tried to plug this into the common test framework so that this will be run for all the supported algorithm combinations.

Let me know if this looks good.